src/components/modals/SupportRequestModal.js

```diff
@@ -0,0 +1,382 @@
+import React, { Component } from 'react';
+import { div, form, input, label, textarea, hh, h, select, span, option, section, button } from 'react-hyperscript-helpers';
+import { Support} from '../../libs/ajax';
+import { Storage } from '../../libs/storage';
+import { Notifications } from '../../libs/utils';
+import { PageSubHeading } from '../PageSubHeading';
+import CloudUploadIcon from '@material-ui/icons/CloudUpload';
+import HighlightOffIcon from '@material-ui/icons/HighlightOff';
+import Dropzone from 'react-dropzone';
+import Modal from 'react-modal';
+
+Modal.setAppElement('#root');
+
+export const SupportRequestModal = hh(
+  class SupportRequestModal extends Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        name: Storage.userIsLogged() ?
+          Storage.getCurrentUser().displayName :
+          '',
+        isLogged: Storage.userIsLogged(),
+        type: 'question',
+        subject: '',
+        description: '',
+        attachment: '',
+        email: Storage.userIsLogged() ? Storage.getCurrentUser().email : '',
+        first_name: Storage.userIsLogged() ?
+          Storage.getCurrentUser().displayName.split(' ')[0] :
+          '',
+        height: Storage.userIsLogged() ? (window.innerHeight < 550 ? window.innerHeight : '550px') : (window.innerHeight < 700 ? window.innerHeight : '700px'),
+        top: Storage.userIsLogged() ? '400px' : '1',
+        valid: Storage.userIsLogged(),
+        validAttachment: true
+      };
+
+      this.closeHandler = () => {
```

Comment:
Why are these methods in the constructor and not just component methods?

Reply:
No real reason. I was following code from pages, I can move them out.